src/titiler/core/titiler/core/factory.py

```diff
@@ -225,6 +233,20 @@ def add_route_dependencies(
                 # https://github.com/tiangolo/fastapi/blob/58ab733f19846b4875c5b79bfb1f4d1cb7f4823f/fastapi/routing.py#L677-L678
                 route.dependencies.extend(dependencies)  # type: ignore
 
+    def add_telemetry(self):
+        """
+        Applies the factory_trace decorator to all registered API routes.
+
+        This method iterates through the router's routes and wraps the endpoint
+        of each APIRoute to ensure consistent OpenTelemetry tracing.
+        """
+        if not factory_trace.decorator_enabled:
```

Comment:
What about having a setting directly in the Factory and then use it in L178 such as
```python
if self.enable_telemetry:
    self.add_telemetry()
```

Reply:
We could do that to provide additional control to the user rather than just turning it on if the telemetry dependencies are installed. I'll add it in.